composition-js/src/validate.ts

```diff
@@ -651,6 +665,16 @@ class ValidationTraversal {
         continue;
       }
 
+      // `state.selectedOverrideConditions` indicates the labels (and their
```

Comment:
Earlier in this function, you'll notice a call to `isSupersetOrEqual(currentSources, previousSources)`; that is the part of the code that stops us from recursing infinitely into the API query graph.

That will need some changes since we're adding new state in the form of `selectedOverrideConditions`. In particular, the current type of `previousVisits` is:
```typescript
previousVisits: QueryGraphState<string[][]>;
```
it would need to change to:
```typescript
previousVisits: QueryGraphState<{ sources: string[], selectedOverrideConditions: Record<string, boolean> }[]>;
```
So you'd push both the `currentSources` and the current `selectedOverrideCondition` onto this instead of just `currentSources`.

The other change is that in addition to `isSupersetOrEqual(currentSources, previousSources)` needing be true to early `return`, it must also be the case that the current `selectedOverrideCondition` is a superset of the previous `selectedOverrideCondition` (with the booleans matching for the labels in common).

Reply:
e8c8a0bb